src/execution/mod.rs

```diff
@@ -431,6 +591,40 @@ where
             mems: memory_instances,
             globals: global_instances,
             data: data_sections,
+            tables,
+            elements,
+            passive_elem_indexes,
         }
     }
 }
+
+fn get_address_offset(value: Value) -> u32 {
+    match value {
+        Value::I32(val) => val,
+        Value::I64(val) => {
+            if val > u32::MAX as u64 {
+                panic!("i64 value for data segment offset is out of reach")
+            }
+            val as u32
+        }
+        // INFO: from wasmtime - implement only global
+        _ => unreachable!(),
+    }
+}
+
+fn read_value_from_wasm(
+    wasm: &[u8],
+    span: &Span,
+    imported_globals: (),
+    imported_funcs: &[FuncInst],
+) -> Value {
+    let mut wasm = WasmReader::new(wasm);
+    wasm.move_start_to(*span).unwrap_validated();
+    let mut stack = Stack::new();
+    run_const(wasm, &mut stack, imported_globals, imported_funcs);
+    let value = stack.peek_unknown_value();
+    if value.is_none() {
```

Comment:
Might want to change this function to return an Error instead of just panic (related to George's previous comments). Maybe a StoreInstantiationError?

Reply:
Might be able to unwrap_validated, check for it